mesonbuild/interpreterbase/interpreterbase.py

```diff
@@ -248,6 +248,8 @@ def resolve_key(key: mparser.BaseNode) -> str:
 
     def evaluate_notstatement(self, cur: mparser.NotNode) -> InterpreterObject:
         v = self.evaluate_statement(cur.value)
+        if v is None:
+            raise InvalidCode('Cannot apply not statement on None value.')
```

Comment:
Technically there is no such thing as a None value, that's a python-level term. In the docs, we call these functions with a void return: https://mesonbuild.com/Reference-manual_elementary_void.html

> Indicates that the function does not return anything.